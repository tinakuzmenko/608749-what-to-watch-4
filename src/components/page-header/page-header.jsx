import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {Pages} from '../../helpers/constants';

const PageHeader = ({isMainPage}) => {
  return (
    <header className="page-header movie-card__head">
      <div className="logo">
        <a
          className="logo__link"
          href={!isMainPage ? `main.html` : null}
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="user-block">
        <div className="user-block__avatar">
          <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
        </div>
      </div>
    </header>
  );
};

PageHeader.propTypes = {
  isMainPage: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isMainPage: state.currentPage === Pages.MAIN,
});

export {PageHeader};
export default connect(mapStateToProps)(PageHeader);
