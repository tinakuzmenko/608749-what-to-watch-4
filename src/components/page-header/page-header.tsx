import * as React from 'react';
import {Link} from 'react-router-dom';
import {connect} from "react-redux";

import {Pages, AuthorizationStatus, AppRoute} from '../../helpers/constants';
import {getAuthorizationStatus, getUserInfo} from '../../store/user/selectors';

const PageHeader = (props) => {
  const {currentPage, isSignedIn, userInfo, children} = props;

  const isSignInPage = currentPage === Pages.SIGN_IN;
  const isMyListPage = currentPage === Pages.MY_LIST;
  const isWithBreadcrumbs = currentPage === Pages.ADD_REVIEW;

  const isWithTitle = isSignInPage || isMyListPage;

  const pageTitleElement = (
    <h1 className="page-title user-page__title">
      {isSignInPage && `Sign in`}
      {isMyListPage && `My list`}
    </h1>
  );

  const userBlockElement = (
    <div className="user-block">
      {isSignedIn &&
        <Link to={AppRoute.MY_LIST}>
          <div className="user-block__avatar">
            <img src={userInfo.avatarUrl} alt={userInfo.name} width="63" height="63" />
          </div>
        </Link>}
      {!isSignedIn &&
        <Link
          to={AppRoute.SIGN_IN}
          className="user-block__link"
        >
        Sign in
        </Link>}
    </div>
  );

  return (
    <header className={`page-header ${isWithTitle ? `user-page__head` : `movie-card__head`}`}>
      <div className="logo">
        <Link
          className="logo__link"
          to={AppRoute.MAIN}
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>
      {isWithBreadcrumbs && children}
      {isSignInPage || isMyListPage ? pageTitleElement : null}
      {!isSignInPage && userBlockElement}
    </header>
  );
};

const mapStateToProps = (state) => ({
  isSignedIn: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
  userInfo: getUserInfo(state),
});

export {PageHeader};
export default connect(mapStateToProps)(PageHeader);
