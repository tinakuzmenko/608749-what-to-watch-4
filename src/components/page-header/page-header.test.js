import React from 'react';
import renderer from 'react-test-renderer';

import {PageHeader} from './page-header';

describe(`PageHeader`, () => {
  it(`Should render correctly when is main page and user signed in`, () => {
    const tree = renderer
      .create(<PageHeader
        isMainPage={true}
        isSignInPage={false}
        isSignedIn={true}
        onSignInClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render correctly when is main page and user is not signed in`, () => {
    const tree = renderer
      .create(<PageHeader
        isMainPage={true}
        isSignInPage={false}
        isSignedIn={false}
        onSignInClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render correctly when is not main page and not sign in page`, () => {
    const tree = renderer
      .create(<PageHeader
        isMainPage={false}
        isSignInPage={false}
        isSignedIn={false}
        onSignInClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render correctly when is sign in page`, () => {
    const tree = renderer
      .create(<PageHeader
        isMainPage={false}
        isSignInPage={true}
        isSignedIn={false}
        onSignInClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
