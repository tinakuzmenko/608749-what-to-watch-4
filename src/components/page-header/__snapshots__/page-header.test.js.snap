// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Should render correctly when is main page 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href={null}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`PageHeader Should render correctly when is not main page 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="main.html"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;
